server, ui: convert debug range page to the admin ui

BramGruneir opened the pull request:
The goal of this change was to try keep the page relatively the same during the port.

Fixes #16583.

![screen shot 2017-08-03 at 16 46 38](https://user-images.githubusercontent.com/1614265/28945515-dc6bdaac-7874-11e7-9d92-907c4a85b38a.png)
![screen shot 2017-08-03 at 16 47 01](https://user-images.githubusercontent.com/1614265/28945520-deea4a98-7874-11e7-8b4c-389bc95f316c.png)
![screen shot 2017-08-03 at 16 47 10](https://user-images.githubusercontent.com/1614265/28945523-e0a7db66-7874-11e7-809d-51c89376517a.png)
![screen shot 2017-08-03 at 16 47 20](https://user-images.githubusercontent.com/1614265/28945525-e26a4b5a-7874-11e7-917b-00957d338a1c.png)
![screen shot 2017-08-03 at 16 47 28](https://user-images.githubusercontent.com/1614265/28945528-e42872e6-7874-11e7-8682-7f26887f35e9.png)
![screen shot 2017-08-03 at 16 47 37](https://user-images.githubusercontent.com/1614265/28945531-e5c0e5f2-7874-11e7-854b-77c2f671accd.png)
![screen shot 2017-08-03 at 16 47 44](https://user-images.githubusercontent.com/1614265/28945533-e77d004c-7874-11e7-8218-becf368564e4.png)

BramGruneir committed:
server, ui: convert debug range page to the admin ui

The goal of this change was to try keep the page relatively the same during the port.

Fixes #16583.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17433)
<!-- Reviewable:end -->

tschottdorf commented:
can you add the time zone at the top of the first screenshot (and make sure it's UTC)?

> Range r1 at ... UTC

BramGruneir commented:
Already done, just wan't in the image.  I took them beforehand.

---

Review status: 0 of 21 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-:-KqfohuaZID4nwAuEe5H:b-sbjnf3)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Some other minor changes since the image was taken.
- The warnings in headers now appear as yellow instead red, for readability and it matches the original.
- Dropped commands counts show up as orange (instead of red) if there is more than one.

couchand commented:
Reviewed 11 of 21 files at r1.
Review status: 11 of 19 files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---

*[pkg/server/status.go, line 846 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxarOB4wfEbm5irB-k:-KqxarOB4wfEbm5irB-l:b-i1v03h) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/server/status.go#L846)):*
> ```Go
> // Range returns rangeInfos for all nodes in the cluster about a specific
> // range. It also returns the range history for that range as well.
> func (s *statusServer) Range(
> ```

I wonder if it would be a good idea to add a test for this?

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 6 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxeCbYnAtbHFLEk4Hc:-KqxeCbYnAtbHFLEk4Hd:b-7jc0as) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L6)):*
> ```XML
> import Long from "long";
> import moment from "moment";
> import { RouterState } from "react-router";
> ```

sort?

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 80 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxeeVIeFpPEmjxXjVj:-KqxeeVJcPArhPwGUp35:b-t7c692) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L80)):*
> ```XML
> // returned as a constant number 0. This converts those constants back into a
> // Long of value 0 or returns the original Long if it already exists.
> function fixLong(value: Long | number): Long {
> ```

maybe put this in `util`?

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 156 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxfjzmF0WxPGUXijv3:-KqxfjzmF0WxPGUXijv4:bmacc9j) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L156)):*
> ```XML
> 
>   // If there is no otherRangeID, it comes back as the number 0.
>   renderRangeID(currentRangeID: Long, otherRangeID: Long) {
> ```

...so should these types be `Long | number`?

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 258 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxgvMN4MOBksw3MBZP:-KqxgvMN4MOBksw3MBZQ:bqbrk29) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L258)):*
> ```XML
>       </div>
>     );
>   }
> ```

You're right, there is a lot here.  This is the point where I'm getting weary reading through this.

The place I'd recommend splitting separate components is at the table level; reading through four distinct table components and then one page combining them is probably reasonable enough.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 278 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kqxhb8V5q7BweWo17Ic:-Kqxhb8W7cdd99Sn-UMV:brp7y21) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L278)):*
> ```XML
>   }
> 
>   getReplicaByID(
> ```

Looks like this isn't used?

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 572 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kqxk96IRHcfkdw2-jOC:-Kqxk96IRHcfkdw2-jOD:b97ax8i) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L572)):*
> ```XML
>       case "statedormant": return {
>         value: ["dormant"],
>         className: ["range-table__cell--raftstate-dormant"],
> ```

I can't help but notice that there's an awful lot of repetition in these cases.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 647 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxlGNYD1a6gdPzzMfv:-KqxlGNYD1a6gdPzzMfw:b-rpt0fr) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L647)):*
> ```XML
>     const sortedStoreIDs = _.chain(infos)
>       .map(info => info.source_store_id)
>       .sort((a, b) => a < b ? -1 : (a > b ? 1 : 0))
> ```

Maybe replicate your comment from below `Use our own function because lodash be broken and will sort lexicographically.`

Or maybe just pull this out into a helper that both can use, eg:

```
const correctlyEvenThoughLodashIsBroken = (a, b) => a < b ? -1 : (a > b ? 1 : 0)

// usage:
// ...
.sort(correctlyEvenThoughLodashIsBroken)
```

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 666 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kqxl8nP4Zvl8WNkQaUk:-Kqxl8nQVxUN9v-g_uEp:b2nifei) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L666)):*
> ```XML
>         dormantStoreIDs.add(info.source_store_id);
>       }
>       detailsByStoreID.set(info.source_store_id, {
> ```

wow

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 715 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kqxm25eDNspo1XV5zFj:-Kqxm25eDNspo1XV5zFk:b3saqtw) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L715)):*
> ```XML
>       .map(rep => rep.replica_id)
>       // Use our own function because lodash be broken and will sort lexicographically.
>       .sort((a, b) => a < b ? -1 : (a > b ? 1 : 0))
> ```

Do you need to sort here, if it's just going straight into a `Set`?

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 719 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxmJd8AOU4gTWwuWsE:-KqxmJd8AOU4gTWwuWsF:b-pu1rnn) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L719)):*
> ```XML
> 
>     // Go through all the replicas and add them to map for easy printing.
>     const replicasByStoreIDByReplicaID: Map<number, Map<number, protos.cockroach.roachpb.ReplicaDescriptor$Properties>> = new Map();
> ```

I *think* the name here is reversed?  Shouldn't the outside map's key be the last "By" clause?

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 842 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kqxn7RHPRjJEBjLeMG7:-Kqxn7RHPRjJEBjLeMG8:bb7r1zc) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L842)):*
> ```XML
> };
> 
> export default connect(mapStateToProps, actions)(Range);
> ```

whew

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433)*
<!-- Sent from Reviewable.io -->

mrtracy commented:
Reviewed 11 of 21 files at r1.
Review status: all files reviewed at latest revision, 19 unresolved discussions, some commit checks failed.

---

*[pkg/server/status.go, line 867 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2JVJDMssK1lEBElsG:-Kr2JVJDMssK1lEBElsH:br97fm0) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/server/status.go#L867)):*
> ```Go
> 
> 	responses := make(chan nodeResponse)
> 	for nodeID := range isLiveMap {
> ```

I wonder if there's *any* way we could abstract this fan-out pattern, which is now being used in at least two (and I think more) admin UI methods.  Maybe there's a way to move the specific parts to a closure of some kind?

I won't hold you to that in this PR, but maybe add a TODO?  The other similar code is in TableStats, I think.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 80 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxeeVIeFpPEmjxXjVj:-Kr2X3kZ5BqRlweCG1nW:biamfdn) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L80)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

maybe put this in `util`?
</blockquote></details>

This doesn't seem correct.  Looking at the generated proto.js, here's the generated default value for a long field:

```
DatabaseDetailsResponse.prototype.descriptor_id = $util.Long ? $util.Long.fromBits(0,0,false) : 0;
```

The check for`$util.Long` is because apparently longjs is optional; however, we are definitely including it in our build.  Optional dependencies often involve some black magic, this is no exception; but it still seems like this should be coming back as a zero long.  We should figure out why it's not and fix that problem.

Even if we accept that "fixLong" is necessary, which I don't, the check should just be `if ((value as any) === 0)`.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 124 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2bHoH2eimRPddtutM:-Kr2bHoH2eimRPddtutN:b-vwgbnw) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L124)):*
> ```XML
>         <table className="connections-table">
>           <tbody>
>             <tr className="connections-table__row connections-table__row--header">
> ```

There is a lot of repetition in the construction of these tables, this seems like something that could and should be abstracted away.  It's making it challenging to find any subtle errors in these code reviews.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 133 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2c4yeyh0ymH9VLhX6:-Kr2c4yeyh0ymH9VLhX7:b-h6adxc) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L133)):*
> ```XML
>               _.map(ids, id => {
>                 const resp = rangeResponse.responses_by_node_id[id];
>                 let headerClassName = "connections-table__row";
> ```

rowClassName.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 304 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2dsgA-CQQ5Wt6JXAu:-Kr2dsgA-CQQ5Wt6JXAv:b-kp2dm3) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L304)):*
> ```XML
>   }
> 
>   printDuration(duration: moment.Duration) {
> ```

This doesn't work for durations longer than 24 hours, it will truncate time.  Should probably be a utility.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 813 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2PaHOL4wwh6VuQPyD:-Kr2PaHOL4wwh6VuQPyE:b-c1gzvs) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L813)):*
> ```XML
>     const replicas = _.chain(infos)
>       .flatMap(info => info.state.state.desc.replicas)
>       .map(rep => rep)
> ```

Is this some sort of typing trick? This appears to be an identity function.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 815 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2Prmub4q0BWXGcnaW:-Kr2Prmub4q0BWXGcnaX:b-dqy6iq) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L815)):*
> ```XML
>       .map(rep => rep)
>       // Use our own function because lodash be broken and will sort lexicographically.
>       .sort((a, b) => a.replica_id < b.replica_id ? -1 : (a.replica_id > b.replica_id ? 1 : 0))
> ```

`sort` is not a lodash function, it is javascript's sort function which sorts lexicographically for odd historical reasons.  At least throw shade at the responsible party!

For numeric sorts, the function can just be `.sort((a, b) => a.replica_id - b.replica_id)`.  If you wanted to leverage lodash, this could also be `_.sortBy((a) => a.replica_id)`, which essentially implements the logic you have here.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 825 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2PHFeBMcAbnP0YJQY:-Kr2PHFeBMcAbnP0YJQZ:b-t0y63v) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L825)):*
> ```XML
>         {this.renderLeaseTable(_.head(infos))}
>         {this.renderLogTable(range.range_id, range.range_log)}
>         {this.renderConnections(range)}
> ```

I would say that at least these four methods should be extracted into components, because each of them has a tremendous number of supporting functions and that has made this class nearly inscrutable.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Replied to @couchand, will update and reply to @mrtracy's comments shortly.

---

Review status: 6 of 26 files reviewed at latest revision, 19 unresolved discussions.

---

*[pkg/server/status.go, line 846 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxarOB4wfEbm5irB-k:-Kr2_AxEZ2pNSyzbiMkK:bz7ngai) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/server/status.go#L846)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

I wonder if it would be a good idea to add a test for this?
</blockquote></details>

Good call.  Done.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 6 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxeCbYnAtbHFLEk4Hc:-Kr1jsI0XJQxzDIX03-2:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L6)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

sort?
</blockquote></details>

Done.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 80 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxeeVIeFpPEmjxXjVj:-Kr1lIgmsLBdsm1SeUmI:b-7rl0jd) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L80)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

maybe put this in `util`?
</blockquote></details>

done.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 156 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxfjzmF0WxPGUXijv3:-Kr1lSuZRN1X1CL0m20W:bpx82t4) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L156)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

...so should these types be `Long | number`?
</blockquote></details>

I guess so.  Typescript doesn't find an error here, since all the protos always contain a Long, even when they are set to number(0).

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 258 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxgvMN4MOBksw3MBZP:-Kr2BnlyvJGe7lT3dDI7:b-q2zz2r) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L258)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

You're right, there is a lot here.  This is the point where I'm getting weary reading through this.

The place I'd recommend splitting separate components is at the table level; reading through four distinct table components and then one page combining them is probably reasonable enough.
</blockquote></details>

Ok, did a fairly large refactoring.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 278 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kqxhb8V5q7BweWo17Ic:-Kr2BrZYS9NH-nfmA76M:b-cp7167) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L278)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

Looks like this isn't used?
</blockquote></details>

it's not, removed.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 572 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kqxk96IRHcfkdw2-jOC:-Kr2ZlRjvwp2ria3fMi7:bdduej1) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L572)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

I can't help but notice that there's an awful lot of repetition in these cases.
</blockquote></details>

I could just strip the `state` from the start and insert it into `range-table__cell--raftstate-{here}`.  But I still want to be able to check to see if it matches the known list of states.  So for that, I could add a regex and return unknown if it doesn't.  But that just seemed like overkill and would be a lot less readable.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 647 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxlGNYD1a6gdPzzMfv:-Kr2XghtWZMaemyN5M-A:b-4x9syw) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L647)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

Maybe replicate your comment from below `Use our own function because lodash be broken and will sort lexicographically.`

Or maybe just pull this out into a helper that both can use, eg:

```
const correctlyEvenThoughLodashIsBroken = (a, b) => a < b ? -1 : (a > b ? 1 : 0)

// usage:
// ...
.sort(correctlyEvenThoughLodashIsBroken)
```
</blockquote></details>

It's only used in one place, but I like pulling out the function.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 666 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kqxl8nP4Zvl8WNkQaUk:-Kr2C26LzUTrs4753N2i:bb7r7k5) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L666)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

wow
</blockquote></details>

wow?

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 715 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kqxm25eDNspo1XV5zFj:-Kr2YnFWx1WW6ym7PFze:b-n1z8vy) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L715)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

Do you need to sort here, if it's just going straight into a `Set`?
</blockquote></details>

good point, it's not needed here anymore.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 719 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxmJd8AOU4gTWwuWsE:-Kr2ZN8fXmKw6IPy2nSY:b-xh4bcr) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L719)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

I *think* the name here is reversed?  Shouldn't the outside map's key be the last "By" clause?
</blockquote></details>

I wasn't sure.  Done.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 842 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kqxn7RHPRjJEBjLeMG7:-Kr2ZSinQmiF_L0FEXcf:b32vhw) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L842)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

whew
</blockquote></details>

:)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-:-Kr2kT-musxnLAqy6a_K:b-fq7gwm)*
<!-- Sent from Reviewable.io -->

couchand commented:
Review status: 6 of 26 files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 156 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxfjzmF0WxPGUXijv3:-Kr6TIEVn702POVCryl7:b-1sbffb) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L156)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

I guess so.  Typescript doesn't find an error here, since all the protos always contain a Long, even when they are set to number(0).
</blockquote></details>

This all seems to be coming from the typings for the library code being wrong and Typescript not diving into it.  Since the ts parts are stripped before the code is ever executed, of course it wouldn't matter if the runtime value is a number.  I think @mrtracy's point above makes sense, and we should keep pulling on this thread all the way up to the source.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 572 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kqxk96IRHcfkdw2-jOC:-Kr6XjiE1CI8ix4F9c4r:b-1oadvu) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L572)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

I could just strip the `state` from the start and insert it into `range-table__cell--raftstate-{here}`.  But I still want to be able to check to see if it matches the known list of states.  So for that, I could add a regex and return unknown if it doesn't.  But that just seemed like overkill and would be a lot less readable.
</blockquote></details>

I was thinking something like:

```
printRaftState(state) {
  switch (state) {
    case "statedormant": return "dormant"
    // ...
    default: return "unknown"
  }
}

contentRaftState(state) {
  str = printRaftState(state)
  return {
    value: [str]
    className: [`range-table__cell--raftstate-${str}`]
  }
}
```

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 666 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kqxl8nP4Zvl8WNkQaUk:-Kr6YS4FWYrzmJ542xq7:b9rrjor) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L666)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

wow?
</blockquote></details>

just this big object of mapped details.  wondering how to make this nicer, but probably not worth digging in to right now.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 815 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2Prmub4q0BWXGcnaW:-Kr6YjDd0SEvU9mr41sI:b-jb7ubg) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L815)):*
<details><summary><i>Previously, mrtracy (Matt Tracy) wrote…</i></summary><blockquote>

`sort` is not a lodash function, it is javascript's sort function which sorts lexicographically for odd historical reasons.  At least throw shade at the responsible party!

For numeric sorts, the function can just be `.sort((a, b) => a.replica_id - b.replica_id)`.  If you wanted to leverage lodash, this could also be `_.sortBy((a) => a.replica_id)`, which essentially implements the logic you have here.
</blockquote></details>

lodash is broken.  punting the blame to JS, when the whole purpose of using a support library is to get around the warts of JS, is just sad.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 6 of 28 files reviewed at latest revision, 12 unresolved discussions.

---

*[pkg/server/status.go, line 867 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2JVJDMssK1lEBElsG:-Kr2lW07SQn8G4W33x7j:b-2rsnbp) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/server/status.go#L867)):*
<details><summary><i>Previously, mrtracy (Matt Tracy) wrote…</i></summary><blockquote>

I wonder if there's *any* way we could abstract this fan-out pattern, which is now being used in at least two (and I think more) admin UI methods.  Maybe there's a way to move the specific parts to a closure of some kind?

I won't hold you to that in this PR, but maybe add a TODO?  The other similar code is in TableStats, I think.
</blockquote></details>

Sure, done.

TableStats seems to use a predetermined subset, not the alive set.  So I left it as is.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 80 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxeeVIeFpPEmjxXjVj:-Kr2nC--m6Xuv8JGASLx:b-l9tfau) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L80)):*
<details><summary><i>Previously, BramGruneir (Bram Gruneir) wrote…</i></summary><blockquote>

done.
</blockquote></details>

Well, I assure you it's broken.  

I've changed it to value as any as you suggested.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 124 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2bHoH2eimRPddtutM:-Kr2nzwJCzjbymOFowXm:btkh1vl) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L124)):*
<details><summary><i>Previously, mrtracy (Matt Tracy) wrote…</i></summary><blockquote>

There is a lot of repetition in the construction of these tables, this seems like something that could and should be abstracted away.  It's making it challenging to find any subtle errors in these code reviews.
</blockquote></details>

I'll all for abstracting more if you feel that's needed.  But this part right here seems pretty straightforward to me.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 133 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2c4yeyh0ymH9VLhX6:-Kr2oOUwter3VuIY1vI_:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L133)):*
<details><summary><i>Previously, mrtracy (Matt Tracy) wrote…</i></summary><blockquote>

rowClassName.
</blockquote></details>

Done.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 156 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KqxfjzmF0WxPGUXijv3:-Kr6Zf1USJVtaGUf7N4x:b-jtiond) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L156)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

This all seems to be coming from the typings for the library code being wrong and Typescript not diving into it.  Since the ts parts are stripped before the code is ever executed, of course it wouldn't matter if the runtime value is a number.  I think @mrtracy's point above makes sense, and we should keep pulling on this thread all the way up to the source.
</blockquote></details>

Sure, but I'd rather just leave the fix/hack in for now and we can dig deeper afterwords.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 304 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2dsgA-CQQ5Wt6JXAu:-Kr2oiyWak9GEbI9TWiz:b-8pkg6s) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L304)):*
<details><summary><i>Previously, mrtracy (Matt Tracy) wrote…</i></summary><blockquote>

This doesn't work for durations longer than 24 hours, it will truncate time.  Should probably be a utility.
</blockquote></details>

Added days.  And yes, it is a utility now.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 572 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kqxk96IRHcfkdw2-jOC:-Kr6_zZkL4TTfSovV5Y9:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L572)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

I was thinking something like:

```
printRaftState(state) {
  switch (state) {
    case "statedormant": return "dormant"
    // ...
    default: return "unknown"
  }
}

contentRaftState(state) {
  str = printRaftState(state)
  return {
    value: [str]
    className: [`range-table__cell--raftstate-${str}`]
  }
}
```
</blockquote></details>

Done.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 813 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2PaHOL4wwh6VuQPyD:-Kr2oomoZn3Oi2RZCZC1:b-o2tqmp) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L813)):*
<details><summary><i>Previously, mrtracy (Matt Tracy) wrote…</i></summary><blockquote>

Is this some sort of typing trick? This appears to be an identity function.
</blockquote></details>

it was erroneous, already removed.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 814 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2pm5LMHvs69ykBolR:-Kr2pm5LMHvs69ykBolS:bngdxz6) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L814)):*
> _.sortBy((a) => a.replica_id)

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 815 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2Prmub4q0BWXGcnaW:-Kr2p3RaJo29GqavCAs3:bvle5qp) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L815)):*
<details><summary><i>Previously, mrtracy (Matt Tracy) wrote…</i></summary><blockquote>

`sort` is not a lodash function, it is javascript's sort function which sorts lexicographically for odd historical reasons.  At least throw shade at the responsible party!

For numeric sorts, the function can just be `.sort((a, b) => a.replica_id - b.replica_id)`.  If you wanted to leverage lodash, this could also be `_.sortBy((a) => a.replica_id)`, which essentially implements the logic you have here.
</blockquote></details>

But .sort is part of lodash's chain in this case, why would they even include it then?  Isn't lodash supposed to fix the rough edges of JS?

Anyway, I've removed the disparaging comment and adjusted it to use lodash's sort.

---

*[pkg/ui/src/views/reports/containers/range/index.tsx, line 825 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-Kr2PHFeBMcAbnP0YJQY:-Kr2p-y3YIJTV2AuQfJ5:brqvn2p) ([raw file](https://github.com/cockroachdb/cockroach/blob/b9a6e8d4a90563fefb8f811bd2fb49974da1ea6c/pkg/ui/src/views/reports/containers/range/index.tsx#L825)):*
<details><summary><i>Previously, mrtracy (Matt Tracy) wrote…</i></summary><blockquote>

I would say that at least these four methods should be extracted into components, because each of them has a tremendous number of supporting functions and that has made this class nearly inscrutable.
</blockquote></details>

yep, already done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433)*
<!-- Sent from Reviewable.io -->

couchand commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> left a number of nits, looks good overall.

---

Reviewed 8 of 24 files at r3.
Review status: 14 of 28 files reviewed at latest revision, 18 unresolved discussions, all commit checks successful.

---

*[pkg/ui/src/views/reports/containers/range/leaseTable.tsx, line 45 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC11OxdbUdm8gVf_JC:-KrC11OxdbUdm8gVf_JD:b1wsmdg) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/leaseTable.tsx#L45)):*
> ```XML
>     }
> 
>     const epoch = Lease.IsEpoch(_.head(info.lease_history));
> ```

%s/epoch/isEpoch/ #nit

---

*[pkg/ui/src/views/reports/containers/range/leaseTable.tsx, line 80 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC1CGqkOgkhOkCWYte:-KrC1CGqkOgkhOkCWYtf:b-8tg6no) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/leaseTable.tsx#L80)):*
> ```XML
>                     {this.renderLeaseCell(Print.TimestampDelta(lease.proposed_ts, prevProposedTimestamp))}
>                     {
>                       !epoch ? this.renderLeaseTimestampCell(lease.expiration) : null
> ```

oneline? #nit

---

*[pkg/ui/src/views/reports/containers/range/logTable.tsx, line 15 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC4iPEtdFcQYa9sOy3:-KrC4iPEtdFcQYa9sOy4:b-cpds2u) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/logTable.tsx#L15)):*
> ```XML
> export default class LogTable extends React.Component<LogTableProps, {}> {
>   // If there is no otherRangeID, it comes back as the number 0.
>   renderRangeID(currentRangeID: Long | number, otherRangeID: Long | number) {
> ```

in this case I'd prefer using the fact that this is a component method, and refer to the current range id as `this.props.rangeID` rather than passing it as a parameter

---

*[pkg/ui/src/views/reports/containers/range/logTable.tsx, line 33 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC4wVixQ_X6DcciIFE:-KrC4wVjleYoo7qjN_Z2:b-romlfr) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/logTable.tsx#L33)):*
> ```XML
>   }
> 
>   printLogEventType(eventType: protos.cockroach.storage.RangeLogEventType) {
> ```

for things that don't rely on the component state I prefer free functions to methods

---

*[pkg/ui/src/views/reports/containers/range/print.ts, line 36 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC1eKWeuE-ScqO1-Pt:-KrC1eKXv-bg1TUrwsFW:b-dtps4) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/print.ts#L36)):*
> ```XML
>   const results: string[] = [];
>   if (duration.days() > 0) {
>     results.push(`${duration.days()}days`);
> ```

maybe s/days/d/ ? as it is, one day will be `1days` which is unfortunate, and "d" seems clear enough in context #nit

---

*[pkg/ui/src/views/reports/containers/range/print.ts, line 67 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC2-C67VJGulfqha4x:-KrC2-C67VJGulfqha4y:byqlmwe) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/print.ts#L67)):*
> ```XML
>   const oldTime = LongToMoment(oldTimestamp.wall_time);
>   const diff = moment.duration(newTime.diff(oldTime));
>   return this.Duration(diff);
> ```

`PrintDuration` in case this is imported like `import {PrintTimestampDelta}`

---

*[pkg/ui/src/views/reports/containers/range/rangeInfo.ts, line 16 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC2RtlrqvDxw2y3Jhy:-KrC2RtlrqvDxw2y3Jhz:b-b99vyp) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/rangeInfo.ts#L16)):*
> ```XML
>     return false;
>   }
>   return Long.fromInt(localRep.replica_id).eq(FixLong(info.raft_state.lead));
> ```

data type question: if the replica id is ok as an int sometimes, why is it ever a long?

---

*[pkg/ui/src/views/reports/containers/range/rangeTable.tsx, line 21 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC2dw9xB4NrYj81s0o:-KrC2dwACL40DoVBWIxt:bcv5izw) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/rangeTable.tsx#L21)):*
> ```XML
>   readonly variable: string;
>   readonly display: string;
>   readonly compareToLeader: boolean; // When true, display a warnings when a value doesn't match the
> ```

s/warnings/warning/ #nit

---

*[pkg/ui/src/views/reports/containers/range/rangeTable.tsx, line 391 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC49PZ-oPAP9iFiOB_:-KrC49PZ-oPAP9iFiOBa:b-cybvvr) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/rangeTable.tsx#L391)):*
> ```XML
>                 sortedStoreIDs,
>                 rangeID,
>                 firstKeyMax + key,
> ```

fyi this could just be `"replicas" + index`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-:-KrC57BFPmJJEAE87AxU:b-u6u8b5)*
<!-- Sent from Reviewable.io -->

BramGruneir commented:
Review status: 10 of 28 files reviewed at latest revision, 18 unresolved discussions.

---

*[pkg/ui/src/views/reports/containers/range/leaseTable.tsx, line 45 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC11OxdbUdm8gVf_JC:-KrCb1hFI50YCzM9jJ4a:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/leaseTable.tsx#L45)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

%s/epoch/isEpoch/ #nit
</blockquote></details>

Done.

---

*[pkg/ui/src/views/reports/containers/range/leaseTable.tsx, line 80 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC1CGqkOgkhOkCWYte:-KrCb5aF8cbsWDpsdFDr:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/leaseTable.tsx#L80)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

oneline? #nit
</blockquote></details>

Done.

---

*[pkg/ui/src/views/reports/containers/range/logTable.tsx, line 15 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC4iPEtdFcQYa9sOy3:-KrCbSZtucpdU8u6vYxA:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/logTable.tsx#L15)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

in this case I'd prefer using the fact that this is a component method, and refer to the current range id as `this.props.rangeID` rather than passing it as a parameter
</blockquote></details>

Done.

---

*[pkg/ui/src/views/reports/containers/range/logTable.tsx, line 33 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC4wVixQ_X6DcciIFE:-KrCbgX5J-vBOXXTEXNt:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/logTable.tsx#L33)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

for things that don't rely on the component state I prefer free functions to methods
</blockquote></details>

Done.

---

*[pkg/ui/src/views/reports/containers/range/print.ts, line 36 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC1eKWeuE-ScqO1-Pt:-KrCcXV5JlW89JrAjbu9:b-8h44ed) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/print.ts#L36)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

maybe s/days/d/ ? as it is, one day will be `1days` which is unfortunate, and "d" seems clear enough in context #nit
</blockquote></details>

`d` is correct in this context, changed

---

*[pkg/ui/src/views/reports/containers/range/print.ts, line 67 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC2-C67VJGulfqha4x:-KrCci3zDXAnnHAWdjdJ:b-odyjl3) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/print.ts#L67)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

`PrintDuration` in case this is imported like `import {PrintTimestampDelta}`
</blockquote></details>

Of course, was a copy-paste (and forget to alter since it was working) error.

---

*[pkg/ui/src/views/reports/containers/range/rangeInfo.ts, line 16 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC2RtlrqvDxw2y3Jhy:-KrCcqy9TVwz63xGgwBF:bknnsv1) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/rangeInfo.ts#L16)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

data type question: if the replica id is ok as an int sometimes, why is it ever a long?
</blockquote></details>

It really shouldn't be an int and we should probably update our protos accordingly.  
It is entirely possible to have replica IDs greater than int32.  However, I don't think even our largest test clusters come even remotely close.

---

*[pkg/ui/src/views/reports/containers/range/rangeTable.tsx, line 21 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC2dw9xB4NrYj81s0o:-KrCbtuLHviAo52yI63a:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/rangeTable.tsx#L21)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

s/warnings/warning/ #nit
</blockquote></details>

Done.

---

*[pkg/ui/src/views/reports/containers/range/rangeTable.tsx, line 391 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-KrC49PZ-oPAP9iFiOB_:-KrCf64feloSRAu_56uj:bn7du0s) ([raw file](https://github.com/cockroachdb/cockroach/blob/5a4c7246736b838c2e299c8a7c2f49ef505e6646/pkg/ui/src/views/reports/containers/range/rangeTable.tsx#L391)):*
<details><summary><i>Previously, couchand (Andrew Couch) wrote…</i></summary><blockquote>

fyi this could just be `"replicas" + index`
</blockquote></details>

Ah, I see what you mean.  Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433)*
<!-- Sent from Reviewable.io -->

mrtracy commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 10 of 28 files reviewed at latest revision, 18 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/17433#-:-KrUMTYH4zfnCplr5HwA:bnfp4nl)*
<!-- Sent from Reviewable.io -->
